icann-rdap-client/src/md/redacted.rs

```diff
@@ -61,3 +68,501 @@ impl ToMd for &[Redacted] {
         md
     }
 }
+
+// These are the different types of results that we can get from the JSON path checks
+#[derive(Debug, PartialEq, Clone)]
+pub enum ResultType {
+    StringNoValue, // (*) what we found in the value paths array was a string but has no value (yes, this is a little weird, but does exist) `Redaction by Empty Value`
+    EmptyString, // (*) what we found in the value paths array was a string but it is an empty string `Redaction by Empty Value`
+    PartialString, // (*) what we found in the value paths array was a string and it does have a value `Redaction by Partial Value` and/or `Redaction by Replacement Value`
+    Array, // what we found in the value paths array was _another_ array (have never found this w/ redactions done correctly)
+    Object, // what we found in the value paths array was an object (have never found this w/ redactions done correctly)
+    Removed, // (*) paths array is empty, finder.find_as_path() found nothing `Redaction by Removal`
+    FoundNull, // value in paths array is null (have never found this w/ redactions done correctly)
+    FoundNothing, // fall through, value in paths array is not anything else (have never found this w/ redactions done correctly)
+    FoundUnknown, // what we found was not a JSON::Value::string (have never found this w/ redactions done correctly)
+    FoundPathReturnedBadValue, // what finder.find_as_path() returned was not a Value::Array (have never found this, could possibly be an error)
+}
+
+#[derive(Debug, Clone)]
+pub struct RedactedObject {
+    pub name: Value,                     // Get the description's name or type
+    pub path_index_count: i32,           // how many paths does the json resolve to?
+    pub pre_path: Option<String>,        // the prePath
+    pub post_path: Option<String>,       // the postPath
+    pub original_path: Option<String>,   // the original path that was put into the redaction
+    pub final_path: Vec<Option<String>>, // a vector of the paths where we put a partialValue or emptyValue
+    pub do_final_path_subsitution: bool, // if we are modifying anything or not
+    pub path_lang: Value, // the path_lang they put in, these may be used in the future
+    pub replacement_path: Option<String>,
+    pub method: Value,                         // the method they are using
+    pub reason: Value,                         // the reason
+    pub result_type: Vec<Option<ResultType>>,  // a vec of our own internal Results we found
+    pub redaction_type: Option<RedactionType>, //
+}
+
+// This isn't just based on the string type that is in the redaction method, but also based on the result type above
+#[derive(Debug, PartialEq, Clone)]
+pub enum RedactionType {
+    EmptyValue,
+    PartialValue,
+    ReplacementValue,
+    Removal,
+    Unknown,
+}
+
+// this is our public entry point
+pub fn replace_redacted_items(orignal_response: RdapResponse) -> RdapResponse {
+    // convert the RdapResponse to a string
```

Comment:
before doing all this, should the code check if there are redactions or there are not redactions and short circuit this if there are no redactions.

Reply:
Good idea, I'll have it check before it calls this and does anything else.